ci/hipsycl.filter

```diff
@@ -37,3 +37,4 @@ vector_load_store
 vector_operators
 vector_swizzle_assignment
 vector_swizzles
+optional_kernel_features
```

Comment:
I think it's better to maintain an alphabetical ordering, the same for other filters. Also, should we maybe name it `kernel_optional_features`?

Reply:
Sorted lines in this commit: acc4949b236617eb437398836aecf32b1c4f102a.

We can rename it but in other PR when all tests will be implemented, because some of the tests in other PR, some of the tests in development and it will be unhandy to rename these tests now.

IMHO, I think we can keep the current name.